### PR TITLE
Feat: Revamped  LHS section in collections & also added some functionality  [#1343]

### PR DESCRIPTION
**Pull request by Abhishek9503**
## Description

#1343 
In this PR 
1. I have merged  the environment section and collection section  in single section named as workspace  
2. Revamped the UI part of lhs section 
3. Integrated the functionality of environment in collection  section 
4. Create global search functionality 
5.  The RHS part for environment is pending will add in the next PR. 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md

### CODE REVIEW

File: src/packages/@library/icons/Stack.svelte [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+<path d="M10.7458 5.52815L6.00009 7.85718L1.25438 5.52815C0.995097 5.40097 0.680812 5.50802 0.553627 5.76828C0.426442 6.02805 0.533003 6.34183 0.793766 6.46901L5.76978 8.91156C5.84246 8.94741 5.92152 8.96509 6.00009 8.96509C6.07866 8.96509 6.15821 8.94741 6.23039 8.91156L11.2064 6.46901C11.4662 6.34183 11.5737 6.02805 11.4465 5.76828C11.3189 5.50851 11.0051 5.40097 10.7458 5.52815Z" fill="#A5B5CA"/>
+<path d="M10.7458 8.06343L6.00009 10.3925L1.25438 8.06343C0.995097 7.93674 0.680812 8.0433 0.553627 8.30356C0.426442 8.56333 0.533003 8.87712 0.793766 9.0043L5.76978 11.4468C5.84246 11.4827 5.92152 11.5004 6.00009 11.5004C6.07866 11.5004 6.15821 11.4827 6.23039 11.4468L11.2064 9.0043C11.4662 8.87712 11.5737 8.56333 11.4465 8.30356C11.3189 8.0438 11.0051 7.93576 10.7458 8.06343Z" fill="#A5B5CA"/>
+</g>
+<defs>
+<clipPath id="clip0_9266_5018">
+<rect width="12" height="12" fill={color}/>
```
Comment (itsmdasifraza) [inline src/packages/@library/icons/Stack.svelte:18]: check color for header env dropdown

**Review by itsmdasifraza — APPROVED** [review]

File: src/packages/@workspaces/common/components/index.ts [hunk 2]
```diff
@@ -4,3 +4,4 @@ export { default as TabularInputFormData } from "./tabular-input/TabularInputFor
 export { default as EnvironmentPicker } from "./environment-picker/EnvironmentPicker.svelte";
 export { default as MissedEnvironment } from "./missed-environment/MissedEnvironment.svelte";
 export { default as ReviewEnvironment } from "./review-environment/ReviewEnvironment.svelte";
+export { default as TabularInputEnvironment } from "./tabular-input/TabularInput.svelte";
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/common/components/index.ts:7]: Rename it to TabularInputV2

File: src/packages/@workspaces/common/hoc/WithButtonEnvironment.svelte [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+   */
+  export let disable;
+  /**
+   * identifies the button is in loading state or not
+   */
+  export let loader;
+  $: ({ onClick, icon, disable, loader } = $$props);
+</script>
+
+<ButtonV2
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/common/hoc/WithButtonEnvironment.svelte:22]: Rename this file in format like  `WithButtonV3`  or `WithButtonV4`

File: src/packages/@workspaces/common/hoc/index.ts [hunk 4]
```diff
@@ -1,2 +1,4 @@
 export { default as WithSelect } from "./WithSelect.svelte";
 export { default as WithButton } from "./WithButton.svelte";
+export { default as WithButtonEnvironment } from "./WithButtonEnvironment.svelte";
+export { default as WithButtonV2 } from "./WithButtonV2.svelte";
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/common/hoc/index.ts:4]: rename `WithButtonEnvironment` to `WithButtonV3`

File: src/packages/@workspaces/features/collection-list/layout/CollectionList.svelte [hunk 5]
```diff
@@ -96,11 +74,14 @@
...
   export let scrollList;
 
-  let runAnimation: boolean = true;
-  let showfilterDropdown: boolean = false;
+  export let searchData: string = "";
+
+  export let isExpandCollection;
+
+  export let toggleExpandCollection;
+
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/collection-list/layout/CollectionList.svelte:82]: not required

File: src/packages/@workspaces/features/collection-list/layout/CollectionList.svelte [hunk 6]
```diff
@@ -186,183 +167,102 @@
...
-          id="filter-btn"
-          class="filter-btn btn bg-backgroundDark d-flex align-items-center justify-content-center p-2 d-none
-          {showfilterDropdown ? 'filter-active' : ''}"
-          style="width: 32px; height:32px; position:relative"
-          on:click={() => (showfilterDropdown = !showfilterDropdown)}
+          style="height: 24px; width:24px;"
+          class="add-icon-container border-0 rounded-1 d-flex justify-content-center align-items-center {isHovered
+            ? 'collections-active'
+            : 'collections-inactive'}"
+          on:click|stopPropagation={showImportCollectionPopup}
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/collection-list/layout/CollectionList.svelte:251]: a check for `guest user `is required as guest user cant access import collection for now

File: src/packages/@workspaces/features/index.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 export * from "./rest-explorer";
 export * from "./save-as-request";
-export * from "./collection-list";
+export * from "./workspace-actions";
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/index.ts:3]: this deletion is not required

File: src/packages/@workspaces/features/workspace-actions/layout/WorkspaceActions.svelte [hunk 8]
```diff
@@ -0,0 +1,592 @@
...
+  import { version } from "../../../../../../src-tauri/tauri.conf.json";
+  import { createDeepCopy } from "$lib/utils/helpers";
+  import { Input } from "@library/forms";
+  import { open } from "@tauri-apps/plugin-shell";
+  import constants from "$lib/utils/constants";
+  import Tooltip from "@library/ui/tooltip/Tooltip.svelte";
+  import MixpanelEvent from "$lib/utils/mixpanel/MixpanelEvent";
+  import { EnvironmentViewModel } from "@app/pages/EnvironmentPage/EnvironmentPage.ViewModel";
+  import EnvironmentList from "@workspaces/features/environment-list/layout/EnvironmentList.svelte";
+  import CollectionList from "@workspaces/features/collection-list/layout/CollectionList.svelte";
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/workspace-actions/layout/WorkspaceActions.svelte:33]: can't import `EnvironmentViewModel` here.
import `EnvironmentList` , `CollectionList` from /features

File: src/packages/@workspaces/features/workspace-actions/layout/WorkspaceActions.svelte [hunk 9]
```diff
@@ -0,0 +1,592 @@
...
+          onDeleteEnvironment={_viewModel.onDeleteEnvironment}
+          onUpdateEnvironment={_viewModel.onUpdateEnvironment}
+          onOpenEnvironment={_viewModel.onOpenEnvironment}
+          onSelectEnvironment={_viewModel.onSelectEnvironment}
+          currentWorkspace={activeWorkspace}
+          environments={$environments}
+          currentEnvironment={$activeEnvironment}
+          {searchData}
+          {toggleExpandEnvironment}
+          bind:isExpandEnvironment
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/workspace-actions/layout/WorkspaceActions.svelte:428]: import this view model to `collectionpage.svelte` and use here using exports

File: src/packages/@library/icons/Stack.svelte (hunk 1 above)
Comment (Abhishek9503) [reply src/packages/@library/icons/Stack.svelte:18]: Resolved

File: src/packages/@workspaces/common/components/index.ts (hunk 2 above)
Comment (Abhishek9503) [reply src/packages/@workspaces/common/components/index.ts:7]: Resovled

File: src/packages/@workspaces/common/hoc/WithButtonEnvironment.svelte (hunk 3 above)
Comment (Abhishek9503) [reply src/packages/@workspaces/common/hoc/WithButtonEnvironment.svelte:22]: Resolved

File: src/packages/@workspaces/features/collection-list/layout/CollectionList.svelte (hunk 6 above)
Comment (Abhishek9503) [reply src/packages/@workspaces/features/collection-list/layout/CollectionList.svelte:251]: Resolved

File: src/packages/@workspaces/features/workspace-actions/layout/WorkspaceActions.svelte (hunk 8 above)
Comment (Abhishek9503) [reply src/packages/@workspaces/features/workspace-actions/layout/WorkspaceActions.svelte:33]: Resovled

File: src/packages/@workspaces/features/workspace-actions/layout/WorkspaceActions.svelte (hunk 9 above)
Comment (Abhishek9503) [reply src/packages/@workspaces/features/workspace-actions/layout/WorkspaceActions.svelte:428]: Resovled

**Pull request merged**